pkg/autoscaler/autoscaler.go

```diff
@@ -108,6 +109,23 @@ func machineAutoscalerResource(targetMachineSet *machinev1.MachineSet, minReplic
 	}
 }
 
+// configMapResource build a new ConfigMap resource.
+func configMapResource(data string) *corev1.ConfigMap {
```

Comment:
You may want to try the builder pattern tooling we've been building for our test suites, see https://github.com/openshift/cluster-api-actuator-pkg/blob/master/testutils/resourcebuilder/core/v1/configmap.go for the configmap version